### PR TITLE
[MP-2564] delete project version endpoint

### PR DESCRIPTION
**Pull request by IslamMonjid**
### What this pull request does
Delete project version endpoint

[MP-2564](https://uniwise.atlassian.net/browse/MP-2564)

[MP-2564]: https://uniwise.atlassian.net/browse/MP-2564?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Graloth — APPROVED** [review]

File: internal/project/service.go [hunk 1]
```diff
@@ -182,3 +183,43 @@ func (s *ServiceImpl) GetProjectVersions(ctx context.Context, projectID int) ([]
...
+
+		return errors.Wrapf(err, "Failed to retrieve project version with ID %d and project ID %d", ID, projectID)
+	}
+
+	return s.deleteProjectVersion(ctx, version)
+}
+
+func (s *ServiceImpl) deleteProjectVersion(ctx context.Context, version *Version) error {
+
+	deleteVersionByIDTransaction, err := s.repo.DeleteVersionByIDTransaction(ctx, version.ID)
```
Comment (Sebsh1) [inline internal/project/service.go:203]:
```suggestion
func (s *ServiceImpl) deleteProjectVersion(ctx context.Context, version *Version) error {
	deleteVersionByIDTransaction, err := s.repo.DeleteVersionByIDTransaction(ctx, version.ID)
```
Unnecessary whitespace

File: internal/rest/v1/project.go [hunk 2]
```diff
@@ -249,3 +249,39 @@ func (h *Handlers) newGetProjectVersionsResponse(
...
+
+	err := h.ProjectService.DeleteProjectVersionByIDAndProjectID(
+		ctx.Request().Context(),
+		req.VersionID,
+		req.ProjectID,
+	)
+	if err != nil {
+		if (err.Error() == "not found") {
+			return echo.ErrNotFound
+		}
```
Comment (Sebsh1) [inline internal/rest/v1/project.go:278]: Instead of checking the error message, check the error with `errors.Is` against `ErrNotFound` that you defined in `internal/project/repository.go`

File: migrations/00002_add_storage_key_to_versions_table.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+			CreatedAt  time.Time
+		}
+
+		type Project struct {
+			ID              uint      `gorm:"primaryKey"`
+			ClientProjectID uint      `gorm:"not null"` // POeditor Project ID
+			Name            string    `gorm:"not null;size:191"`
+			Versions        []Version `gorm:"constraint:OnDelete:CASCADE;"`
+			CreatedAt       time.Time
+		}
```
Comment (Sebsh1) [inline migrations/00002_add_storage_key_to_versions_table.go:27]: This migration modifies the `Project` struct, but there are no changes to it in `internal/project/entity.go`?

**Review by Sebsh1 — CHANGES_REQUESTED** [review]

**Review by nikolai-str — APPROVED** [review]

**Review by Sebsh1 — APPROVED** [review]
